script/testing/util/periodic_task.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/python3
+from threading import Timer
+
+
+class PeriodicTask:
+    """
+    A utility class for simple periodic tasks with Timer threads
+    """
+    def __init__(self, interval, function, *args, **kwargs):
+        """
+        Initialize the PeriodicTask object and schedule the initial Timer
+        """
+        self._timer = None
+        self.function = function
+        self.interval = interval
+        self.args = args
+        self.kwargs = kwargs
+        self.is_running = False
+        self.start()
```

Comment:
I would think we want to separate the initialization and the start command so that we can initialize and then start when we see fit.

Reply:
Yes we can, but I think this one is more accurate in the face of time drift. Because every line of code takes time to execute, the 5 sec would be 5 sec + time needed for spawning the thread in real runtime. I tried a number of ways to see which resulted in the least time drift, this method won by only incurring like 10+ ms drift.

Reply:
Really I would have thought something like this would have had less latency between the start and measuring 
```
            self.collect_mem_thread = PeriodicTask(
                self.collect_mem_freq, update_mem_info, self.db_process.pid,
                self.collect_mem_freq, test_case.mem_metrics.mem_info_dict)

        with open(test_case.test_output_file, "w+") as test_output_fd:
            ret_val, _, _ = run_command(test_case.test_command,
                                        test_case.test_error_msg,
                                        stdout=test_output_fd,
                                        stderr=test_output_fd,
                                        cwd=test_case.test_command_cwd)

        self.collect_mem_thread.start() # or is the run_command blocking I forgot. If it is this definitely doesn't work.
```

Reply:
No this would not work. Because the `run_command` is blocking. We need to `start` the thread before the `run_command`